tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/PostRepositoryTest.java

```diff
@@ -172,6 +173,61 @@ public void findPostsAndPlacementsByProgrammeIdAndSpecialtyIdShouldFindPostsLink
               .getForenames()));
 
     }
+  }
+
+  @Test
+  public void testClearOldPostReferencesInPostTableWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
+    entityManager.persist(post2);
+    entityManager.flush();
+
+    postRepository.clearOldPostReferences(post1.getId());
+    entityManager.flush();
+    entityManager.clear();
+
+    Post updatedPost = entityManager.find(Post.class, post2.getId());
+    Assert.assertNull(updatedPost.getOldPost());
+  }
 
+  @Test
+  public void testClearNewPostReferencesInPostTableWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setNewPost(post1);
+    entityManager.persist(post2);
+    entityManager.flush();
+
+    postRepository.clearNewPostReferences(post1.getId());
+    entityManager.flush();
+    entityManager.clear();
+
+    Post updatedPost = entityManager.find(Post.class, post2.getId());
+    assertNull(updatedPost.getNewPost());
+  }
+
+  @Test
+  public void testClearPostReferencesForBothOldAndNewPostWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
+    post1.setNewPost(post2);
+    entityManager.persist(post2);
+    entityManager.flush();
+
+    postRepository.clearPostReferences(post1.getId());
+    entityManager.flush();
+    entityManager.clear();
+
+    Post updatedPost = entityManager.find(Post.class, post2.getId());
+    assertNull(updatedPost.getOldPost());
+    assertNull(updatedPost.getNewPost());
```

Comment:
I don't quite understand what this assertion verifies.

I would suggest having 3 posts: `post1, post2, post3`, and then set values like below:
```
post1.setNewPost(post2)
post2.setOldPost(post1);
post2.setNewPost(post3);
post3.setOldPost(post2);
```
So you can verify if post2 is deleted, the reference in post1 and post3 can be removed.

Besides, I also have another question. In the case I wrote above, do you need to link post1 & post3?

Reply:
Action taken